src/ui/utils/testRuns.ts

```diff
@@ -0,0 +1,72 @@
+import orderBy from "lodash/orderBy";
+
+import { Recording } from "shared/graphql/types";
+
+function testPassed(recording: Recording) {
+  return recording.metadata?.test?.result === "passed";
+}
+
+function testFailed(recording: Recording) {
+  return (
+    recording.metadata?.test?.result &&
+    ["failed", "timedOut"].includes(recording.metadata?.test?.result)
+  );
+}
+export function groupRecordings(recordings: Recording[] | undefined) {
+  const passedRecordings: Recording[] = [];
+  const failedRecordings: Recording[] = [];
+  const flakyRecordings: Recording[] = [];
+  const sortedRecordings = orderBy(recordings, "date", "desc");
+  const flakySet = new Set<string>();
+
+  const fileCounts = sortedRecordings.reduce((acc, recording) => {
+    const file = recording.metadata?.test?.file;
+    if (!file) {
+      return acc;
+    }
+    if (!acc[file]) {
+      acc[file] = 0;
+    }
+    acc[file]++;
+    return acc;
+  }, {} as Record<string, number>);
+
+  for (const recording of sortedRecordings) {
+    const file = recording.metadata?.test?.file;
+
+    if (!file) {
+      continue;
+    }
+    if (fileCounts[file] > 1) {
+      const recordings = sortedRecordings.filter(r => r.metadata?.test?.file === file);
+      const passed = recordings.filter(r => testPassed(r));
+      const failed = recordings.filter(r => testFailed(r));
+
+      // We currently only show the first failed recording.
+      // What we'd eally want to do is show all the recordings as a group
+      if (passed.length > 0 && failed.length > 0) {
+        flakyRecordings.push(recording);
+        flakySet.add(file);
+      }
+
+      if (passed.length > 0) {
+        passedRecordings.push(...passed);
+      } else if (failed.length > 0) {
+        failedRecordings.push(failed[0]);
+      }
+    } else {
+      if (testPassed(recording)) {
+        passedRecordings.push(recording);
+      } else if (testFailed(recording)) {
+        failedRecordings.push(recording);
+      }
+    }
+  }
+
+  return {
+    passedRecordings,
+    failedRecordings,
+    flakyRecordings,
+    flakyCount: flakySet.size,
+  };
```

Comment:
I think this hook might want to be a bit more prescriptive about the display counts. Unless I'm reading it wrong, right now we'll include flaky tests in both the flaky count and the failed count (because failed count is just `failedRecordings.length`)

Seems like we _don't_ want to do show that on the side bar.

Reply:
We definitely don't want to double count. I think I was missing an `else if`.